core/src/main/java/org/apache/gravitino/storage/relational/service/MetadataObjectService.java

```diff
@@ -69,101 +70,130 @@ public static long getMetadataObjectId(
-      if (topicPO == null) {
-        return null;
-      }
-
-      String schemaName = getSchemaFullName(topicPO.getSchemaId());
-      if (schemaName == null) {
-        return null;
-      }
-
-      return DOT_JOINER.join(schemaName, topicPO.getTopicName());
+    long tableId =
+        TableMetaService.getInstance().getTableIdBySchemaIdAndName(schemaId, names.get(2));
+    if (type == MetadataObject.Type.TABLE) {
+      return tableId;
     }
 
-    if (metadatatype == MetadataObject.Type.FILESET) {
-      FilesetPO filesetPO = FilesetMetaService.getInstance().getFilesetPOById(metadataObjectId);
-      if (filesetPO == null) {
-        return null;
-      }
-
-      String schemaName = getSchemaFullName(filesetPO.getSchemaId());
-      if (schemaName == null) {
-        return null;
-      }
-
-      return DOT_JOINER.join(schemaName, filesetPO.getFilesetName());
+    if (type == MetadataObject.Type.COLUMN) {
+      return TableColumnMetaService.getInstance()
+          .getColumnIdByTableIdAndName(tableId, names.get(3));
     }
 
-    throw new IllegalArgumentException(String.format("Doesn't support the type %s", metadatatype));
-  }
-
-  @Nullable
-  private static String getCatalogFullName(Long entityId) {
-    CatalogPO catalogPO = CatalogMetaService.getInstance().getCatalogPOById(entityId);
-    if (catalogPO == null) {
-      return null;
-    }
-    return catalogPO.getCatalogName();
+    throw new IllegalArgumentException(String.format("Doesn't support the type %s", type));
   }
 
+  // Metadata object may be null because the metadata object can be deleted asynchronously.
   @Nullable
-  private static String getSchemaFullName(Long entityId) {
-    SchemaPO schemaPO = SchemaMetaService.getInstance().getSchemaPOById(entityId);
-
-    if (schemaPO == null) {
-      return null;
-    }
-
-    String catalogName = getCatalogFullName(schemaPO.getCatalogId());
-    if (catalogName == null) {
-      return null;
-    }
+  public static String getMetadataObjectFullName(String type, long metadataObjectId) {
+    MetadataObject.Type metadataType = MetadataObject.Type.valueOf(type);
+    String fullName = null;
+    long objectId = metadataObjectId;
+
+    do {
+      switch (metadataType) {
+        case METALAKE:
+          MetalakePO metalakePO = MetalakeMetaService.getInstance().getMetalakePOById(objectId);
+          if (metalakePO != null) {
+            fullName = metalakePO.getMetalakeName();
+            metadataType = null;
+          } else {
+            return null;
+          }
+          break;
+
+        case CATALOG:
+          CatalogPO catalogPO = CatalogMetaService.getInstance().getCatalogPOById(objectId);
+          if (catalogPO != null) {
+            fullName =
+                fullName != null
+                    ? DOT_JOINER.join(catalogPO.getCatalogName(), fullName)
+                    : catalogPO.getCatalogName();
+            metadataType = null;
```

Comment:
Don't change to `MetadataObject.Type.METALAKE` to obtain the metalake name?

Reply:
No, type "CATALOG" is a top-level metadata object, we don't need to get metalake name.